core/sail/shacl/src/main/java/org/eclipse/rdf4j/sail/shacl/ShaclSailConnection.java

```diff
@@ -103,6 +117,13 @@
 		this.shapesRepoConnection = shapesRepoConnection;
 		this.sail = sail;
 		this.transactionSettings = getDefaultSettings(sail);
+		this.useDefaultShapesGraph = sail.getShapesGraphs().contains(RDF4J.SHACL_SHAPE_GRAPH);
+		this.shapesGraphs = sail.getShapesGraphs().stream().map(g -> {
+			if (g.equals(RDF4J.NIL)) {
+				return null;
+			}
+			return g;
+		}).toArray(IRI[]::new);
```

Comment:
This isn't really optimal since users could update the shapes graph set in the ShaclSail and only new connections would take them into consideration.